src/duckdb.mjs

```diff
@@ -1,361 +1,234 @@
+  }
 
-      return this;
-    }
+  async describeTables() {
+    const tables = await this.query(`SHOW TABLES`);
+    return tables.map(({name}) => ({name}));
+  }
 
-    async insertParquet(name, buffer) {
-      const db = await this.db();
-      await db.registerFileBuffer(name, new Uint8Array(buffer));
-      const conn = await db.connect();
-      await conn.query(
-        `CREATE VIEW '${name}' AS SELECT * FROM parquet_scan('${name}')`
-      );
-      await conn.close();
+  async describeColumns({table} = {}) {
+    const columns = await this.query(`DESCRIBE ${table}`);
+    return columns.map(({column_name, column_type}) => {
+      return {
+        name: column_name,
+        type: getType(column_type),
+        databaseType: column_type
+      };
+    });
+  }
 
-      return this;
+  static async of(sources = {}, config = {}) {
+    const db = await createDuckDB();
+    if (config.query?.castTimestampToDate === undefined) {
+      config = {...config, query: {...config.query, castTimestampToDate: true}};
     }
+    await db.open(config);
+    await Promise.all(
+      Object.entries(sources).map(async ([name, source]) => {
+        if ("array" in source) { // array + options
+          const {array, ...options} = source;
+          await insertArray(db, name, array, options);
+        } else if ("file" in source) { // file + options
+          const {file, ...options} = source;
+          await insertFile(db, name, file, options);
+        } else if (source instanceof FileAttachment) { // bare file
+          await insertFile(db, name, source);
+        } else if (Array.isArray(source)) { // bare data
+          await insertArray(db, name, source);
+        } else {
+          throw new Error(`invalid source: ${source}`);
+        }
+      })
+    );
+    return new DuckDBClient(db);
+  }
+}
 
-    async insertArrowTable(name, table, options) {
-      const buffer = arrow.tableToIPC(table);
-      return this.insertArrowFromIPCStream(name, buffer, options);
+async function insertFile(database, name, file, options) {
+  const url = await file.url();
+  if (url.startsWith("blob:")) {
+    const buffer = await file.arrayBuffer();
+    await database.registerFileBuffer(file.name, new Uint8Array(buffer));
+  } else {
+    await database.registerFileURL(file.name, url);
+  }
+  const connection = await database.connect();
+  try {
+    switch (file.mimeType) {
+      case "text/csv":
+        await connection.insertCSVFromPath(file.name, {
+          name,
+          schema: "main",
+          ...options
+        });
+        break;
+      case "application/json":
+        await connection.insertJSONFromPath(file.name, {
+          name,
+          schema: "main",
+          ...options
+        });
+        break;
+      default:
+        if (file.name.endsWith(".parquet")) {
+          await connection.query(
+            `CREATE VIEW '${name}' AS SELECT * FROM parquet_scan('${file.name}')`
+          );
+        } else {
```

Comment:
Add a case for `file.name.endsWith(".arrow")`

Reply:
Added in 71fbda3.